cmd/thanos/store.go

```diff
@@ -379,6 +382,25 @@ func runStore(
 			cancel()
 		})
 	}
+
+	infoSrv := info.NewInfoServer(
+		component.Store.String(),
+		func() []labelpb.ZLabelSet {
+			return bs.LabelSet()
+		},
+		func() *infopb.StoreInfo {
+			mint, maxt := bs.TimeRange()
+			return &infopb.StoreInfo{
+				MinTime: mint,
+				MaxTime: maxt,
+			}
+		},
+		func() *infopb.ExemplarsInfo { return nil },
```

Comment:
Can we use `nil` here? What's the difference between nil and a function returning nil?

Reply:
Sorry missed it. Yes, we can pass `nil` as we are doing with other parameters. We are allowing to pass `nil` for convenience. https://github.com/thanos-io/thanos/pull/4282#discussion_r641449154